mesonbuild/mesonlib/universal.py

```diff
@@ -350,9 +350,8 @@ def perms_s_to_bits(cls, perms_s: T.Optional[str]) -> int:
         return perms
 
 dot_C_dot_H_warning = """You are using .C or .H files in your project. This is deprecated.
-         Currently, Meson treats this as C code, but this
-            might change in the future, breaking your build.
-         You code also might be already broken on gcc and clang.
+         Currently, Meson treats this as C++ code, but they
+            used to be treated as C code.
```

Comment:
This should be extended with something like "The Visual Studio compiler will always treat files with a .C extension as plain C, whereas GCC and Clang will compile them as C++. The compiler flag /TP can be used to manually force Visual Studio to compile all target sources as C++ instead. This is unreliable, so you should consider changing the file extension to an unambiguous one instead."

Reply:
Done